Add Cross-App Access (XAA) setup documentation for Okta (#38478)
DataDog/documentation: content/en/account_management/org_settings/cross_app_access.md

---
title: Cross-App Access
description: Configure Okta Cross-App Access so AI agents can call the Datadog API on behalf of users authorized in Okta.
algolia:
  tags: ["cross-app access", "XAA", "Okta", "AI agent", "MCP", "ID-JAG"]
further_reading:
    - link: '/account_management/org_settings/mobile_third_party_access/'
      tag: 'Documentation'
      text: 'Mobile and Third-Party Access'
    - link: '/account_management/saml/'
      tag: 'Documentation'
      text: 'Configure SAML single sign-on'
    - link: '/mcp_server/'
      tag: 'Documentation'
      text: 'Datadog MCP Server'
---

<div class="alert alert-info">Cross-App Access is in Preview.</div>

## Overview

Cross-App Access (XAA) lets AI agents call the Datadog API on behalf of users your organization already authorized in Okta. Without it, every user authorizes each agent individually through a browser consent screen. With it, your Okta administrator grants that access once, centrally, and users skip the per-user consent step.

> Okta issues each agent a short-lived token called an ID-JAG (Identity Assertion JWT Authorization Grant). The agent presents this token to Datadog, and Datadog exchanges it for an access token scoped to the user Okta named. Because Okta mints the token, your administrators control which agents reach Datadog and revoke that access from Okta.

Reviewer: ```suggestion
Okta issues each agent a short-lived token called an ID-JAG (Identity Assertion JWT Authorization Grant). The agent presents this token to Datadog, and Datadog exchanges it for an access token scoped to the user Okta named. Because Okta mints the token, your administrators grant and revoke Datadog access for AI agents from Okta.
```

/nit something about this doesn't read quite right 

> Because Okta mints the token, your administrators control which agents reach Datadog and revoke that access from Okta.

Author: Applied your suggestion verbatim. Reads better, thanks.

2. Click **Enable**. This applies to your whole organization.
3. In **Okta issuer URL**, enter the issuer URL for your Okta tenant, for example `https://<YOUR_OKTA_SUBDOMAIN>.okta.com`. Click **Save**.
4. Copy the **Public Org ID**. You enter this in Okta.
5. Copy the client ID for each agent listed under **Registered client IDs**. You enter these in Okta.

{{< img src="account_management/cross_app_access/cross-app-access-settings.png" alt="The Cross-App Access page in Datadog organization settings, showing the enablement status, the Okta issuer URL field, the Public Org ID, and a table of registered client IDs for Claude and Cursor" style="width:100%;" >}}

Datadog derives the token signing keys location from the Okta issuer URL, so the value must be exact:

- Use `https`.
- Enter the bare host with no path, port, query string, fragment, username, or password.
- Use a subdomain of `.okta.com`, `.oktapreview.com`, or `.okta-emea.com`. Datadog rejects the apex domain, so `example.okta.com` works and `okta.com` does not.

Clearing the field and saving removes the issuer and stops Datadog from accepting tokens.

## Control what an agent can do

> 

Reviewer: I think we can be more explicit in saying that although you can do the scopes restrictions in either place (ddog or okta) - you only _need_ to pick 1 if you want the restrictions to apply & we suggest ddog

Author: Rewritten. The section now leads with: you can restrict in either place, both take effect, so you only need to set it in one - and then gives the reasons we recommend Datadog. The Okta **Scope Condition** row is now conditional too: "**Allow all**, if you control scopes in Datadog ... Otherwise, list the scopes this agent is allowed."

      text: 'Mobile and Third-Party Access'
    - link: '/account_management/saml/'
      tag: 'Documentation'
      text: 'Configure SAML single sign-on'
---

{{< callout url="#" btn_hidden="true" header="false">}}
  Cross-App Access is in Preview. Okta manages access to the preview and enables it for your Okta tenant. Any Datadog organization can enable Cross-App Access on the Datadog side.
{{< /callout >}}

## Overview

Cross-App Access (XAA) lets AI agents call the Datadog API on behalf of users your organization already authorized in Okta. Without it, every user authorizes the agent individually through a browser consent screen. With it, your Okta administrator grants that access once, centrally, and users skip the per-user consent step.

Okta issues the agent a short-lived token called an ID-JAG (Identity Assertion JWT Authorization Grant). The agent presents this token to Datadog, and Datadog exchanges it for an access token owned by the user who initiated the call. Because Okta mints the token, your administrators grant and revoke Datadog access for AI agents from Okta.

In Preview, Cross-App Access supports Okta as the only identity provider and Claude as the only agent.

> ## How it works

Reviewer 2: I'd recommend cutting this section. It mostly restates some of the text in Overview, and it pushes the important prereq and setup content further down on the page, which is what a user is actually looking for.

<tr><td><strong>Audience/tenant ID</strong></td><td>Your Datadog organization UUID</td></tr>
</tbody>
</table>
{{< /site-region >}}

{{< site-region region="gov,gov2" >}}
<div class="alert alert-danger">Cross-App Access is not supported for your selected <a href="/getting_started/site/">Datadog site</a> ({{< region-param key="dd_site_name" >}}).</div>
{{< /site-region >}}

The issuer URL identifies the Datadog authorization server, not the token endpoint. Okta writes it into the `aud` claim of the tokens it issues, and Datadog accepts a token only when that claim matches.

**Note**: Changing the issuer URL later requires deleting and recreating the resource connection described in [Connect Claude to the Datadog application](#connect-claude-to-the-datadog-application).

### Register Claude as an AI Agent

Create an AI Agent entry for Claude in Okta, then exchange keys with Anthropic. Anthropic signs the requests Okta receives, so Okta needs Anthropic's public key before it issues any token.

1. Create the AI Agent entry for Claude.
2. Assign owners to the agent. Okta requires an owner before you can activate it.
3. Send the AI Agent ID that Okta generates to Anthropic.

> 4. Add the public key that Anthropic returns to the AI Agent entry, on the **Credentials** tab.

Reviewer 2: Please wrap UI elements like buttons and tabs in this `ui` tag. This prevents our translation service from trying to translate UI strings.
```suggestion
4. Add the public key that Anthropic returns to the AI Agent entry, on the {{< ui >}}Credentials{{< /ui >}} tab.
```

Until the public key is in place, token exchange fails even though every other value is correct. This exchange is manual, so start it early.

### Connect Claude to the Datadog application

On the Claude AI Agent, add the Claude SAML application as a delegated caller, then connect the agent to your Datadog application.

1. On the **Delegations** tab, add the Claude SAML application as a caller.
2. On the **Resource connections** tab, add a resource connection. Select **Application** as the resource type, then select your Datadog application.
3. Set the following fields.

  | Okta field                | Value                                                                                                |
  | ------------------------- | ---------------------------------------------------------------------------------------------------- |
  | **Client ID at resource** | `391e6845-8153-4de1-bbf0-c1b6ef7fdc14`                                                               |
  | **Scope Condition**       | **Allow all**, the only supported value. See [Control scopes in Datadog](#control-scopes-in-datadog) |

4. Activate the agent from the **Actions** menu.

> ### Control scopes in Datadog

Reviewer 2: This section is currently nested under "Finish the setup in Okta," but it is Datadog config. Consider promoting this to h2 or moving it under "Configure cross-app access in Datadog."

**Allow all** is the only supported **Scope Condition** for Cross-App Access. Set it in Okta, then restrict what Claude reaches from Datadog.

Okta does not filter scopes. With **Allow all**, Okta copies whatever Claude requests into the token, which makes Datadog the enforcement point.

<div class="alert alert-warning">Do not enter a list of scopes in Okta. Okta rejects any token request that contains a scope outside the list, so the integration fails with an error instead of falling back to narrower access.</div>

To set the scopes Claude is allowed:

1. Navigate to [**Organization Settings > Mobile and Third-Party Access**](https://app.datadoghq.com/organization-settings/mobile-third-party-access).
2. Select the Claude application, then select the **Scopes** tab.
3. Use the **Allowed** checkbox for each scope to control what Claude reaches.
4. Click **Enable** to save.

Adding or removing a scope affects every user in your organization, and removing a scope revokes existing authorizations that rely on it. See [Application Scope Management](/account_management/org_settings/mobile_third_party_access/#application-scope-management).

> A scope that is not allowed in Datadog is never granted, whatever the token requests.

Reviewer 2: ```suggestion
A scope that is not allowed in Datadog is never granted, regardless of what the token requests.
```

## Configure Cross-App Access in Datadog

Complete the Datadog steps before the Okta steps. Datadog rejects tokens for organizations that have not enabled Cross-App Access, so configuring Okta first produces failures until you finish here.

### Enable Cross-App Access

Set the `mcp_cross_app_access_enabled` org config to `true`. This applies to your whole organization.

```shell
curl -X PATCH "{{< region-param key="dd_api" >}}/api/v2/org_configs/mcp_cross_app_access_enabled" \
  -H "Content-Type: application/json" \
  -H "Authorization: Bearer ${DD_TOKEN}" \
  -d '{
    "data": {
      "type": "org_configs",
      "attributes": {
        "value": true
      }
    }
  }'
```

To turn Cross-App Access off later, send the same request with `"value": false`.

### Set your Okta issuer URL

Datadog derives the token signing keys location from this value, so it must be exact.

```shell
curl -X PUT "{{< region-param key="dd_api" >}}/api/v2/login/org_configs/mcp_cross_app_access_issuer_url" \
  -H "Content-Type: application/json" \
  -H "Authorization: Bearer ${DD_TOKEN}" \
  -d '{
    "data": {

>       "type": "org_config",

Author: It's a good callout, but this is what we have today in our API documentation. I'lll ignore for now.
